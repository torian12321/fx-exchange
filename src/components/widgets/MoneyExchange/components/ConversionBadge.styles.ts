import { css } from '@emotion/core'
import { theme, functions } from 'assets/styles';
// import { transparentize } from 'assets/styles/functions';

// const color = theme.colors.bg;

export const wrapper = css`
  position: relative;
  margin: 1em -1em;
  text-align: center;
`;
export const progressBar = css`
  position: absolute;
  width: 100%;
  left: 0;
  height: 2px;
  top: 50%;
  background: ${theme.colors.primary};
`;

export const badge = css`
  // font-size: .8em;
  position: relative;
  display: inline;
  background: ${theme.colors.bgDark};
  color: ${functions.smartTextColor(theme.colors.bgDark)};

  padding: .3em 1em;
  border-radius: 6em;

  transition: 
    height ${theme.transitionTime} ease-out,
    opacity ${theme.transitionTime} ease 0.5s;

  & > * {
    display: inline-block;
  }
`;
export const badgeHighlighted = css`
  background: ${theme.colors.secondary};
  color: ${functions.smartTextColor(theme.colors.secondary)};
`;

export const icon = css`
  position: relative;
  top: .2em;
  width: 1em;
  margin-right: 0.8em;
`;
export const text = css`
`;
