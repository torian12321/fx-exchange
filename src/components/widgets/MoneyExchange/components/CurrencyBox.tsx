import React, { useState, useEffect} from "react";
/** @jsx jsx */
import { jsx } from '@emotion/core'
import { useCurrenciesState } from 'apiContext';
// import * as styles from "./ConversionBadge.styles";
// import { Icon } from 'components/ui';

export interface IProps {
  currencyCode: string;
}

const CurrencyBox = ({ currencyCode }: IProps) => {
  const { getCurrencyById, lastUpdate } = useCurrenciesState();
  const [currData, setCurrData]: [any, any] = useState();

  useEffect(() => {
    // Get up to date Data.
    setCurrData(getCurrencyById(currencyCode));
  }, [currencyCode, lastUpdate]);

  return (
    <div>
      <span>bla bla {currencyCode}</span>
    </div>
  );
}

export { CurrencyBox };
