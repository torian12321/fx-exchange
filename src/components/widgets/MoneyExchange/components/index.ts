export { ConversionBadge } from './ConversionBadge';
export { MoneySelector } from './MoneySelector';
