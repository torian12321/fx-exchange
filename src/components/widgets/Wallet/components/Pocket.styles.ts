import { css } from '@emotion/core'
import { theme, functions } from 'assets/styles';

const color = theme.colors.bgLight;

export const wrapper = css`
  display: inline-flex;
  width: 100%;
  font-size: 1rem;
  position: relative;
  padding: .6em 1em;
  margin: 0 0 .5em;
  border-radius: ${theme.border.radius};
  box-sizing: border-box;
  line-height: 2em;

  border: ${theme.border.border};
  background: ${color};
  color: ${functions.smartTextColor(color)};
`;

export const image = css`
  font-size: 2em;
  width: 1em;
  height: 1em;
`;
export const label = css`
  font-weight: bold;
  margin: 0 1em;
`;
