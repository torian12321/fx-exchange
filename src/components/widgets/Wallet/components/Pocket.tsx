import React from "react";
/** @jsx jsx */
import { jsx } from '@emotion/core'
import * as styles from "./Pocket.styles";

export interface IProps {
  value: any;
}

const Pocket = ({ value = 0 }: IProps) => {
  const { img = '', symbol = '' } = value;

  return (
    <div css={styles.wrapper}>
      <img src={img} css={styles.image} />
      <span css={styles.label}>
        {`${value.value} ${symbol}`}
      </span>
    </div>
  )
};

export { Pocket };
