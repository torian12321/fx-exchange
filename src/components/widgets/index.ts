export { WidgetMoneyExchange } from './MoneyExchange';
export { WidgetWallet } from './Wallet';
