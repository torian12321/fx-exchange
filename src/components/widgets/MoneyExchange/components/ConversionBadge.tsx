import React from "react";
/** @jsx jsx */
import { jsx } from '@emotion/core'
import * as styles from "./ConversionBadge.styles";
import { Icon } from 'components/ui';
import iconSrc from 'assets/img/icons/profit.svg';

export interface IProps {
  value: string | number;
}

const ConversionBadge = ({ value }: IProps) => (
  <div css={styles.badge} >
    <Icon src={iconSrc} />
    <span>{ value }</span>
  </div>
);

export { ConversionBadge };
