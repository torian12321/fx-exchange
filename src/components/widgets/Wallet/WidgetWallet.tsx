import React, { useState } from 'react';
/** @jsx jsx */
import { jsx } from '@emotion/core';
import { WalletProvider, useWalletState } from 'apiContext';
import { Pocket } from './components/Pocket';
import * as styles from "./WidgetWallet.styles";

export const WidgetWallet = () => {
  const { currencies } = useWalletState();

  return (
    <div css={styles.wrapper}>
      {currencies.map((c: any) => <Pocket value={c} key={c.id} />)}
    </div>
  )
}

export const WidgetWalletStoryBook = () => (
  <WalletProvider>
    <WidgetWallet />
  </WalletProvider>
);
