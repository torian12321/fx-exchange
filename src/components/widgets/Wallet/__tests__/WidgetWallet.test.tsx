import React from 'react';
import { render } from '@testing-library/react';
import { create } from "react-test-renderer";
import '@testing-library/jest-dom/extend-expect';
import { WalletProvider } from 'apiContext';
import { WidgetWallet } from '../WidgetWallet';
import { Pocket } from '../components/Pocket';


describe("Widget: Wallet", () => {
  it(`should render no items by default`, () => {
    // Given && When
    const wallet = create(
      <WidgetWallet />
    );

    // Then
    expect(wallet.toJSON()).toMatchSnapshot();
    expect(wallet.root.findAllByType(Pocket).length).toEqual(0);
  });

  it(`should render with 3 currencies from the 'API'`, () => {
    // Given && When
    const wallet = create(
      <WalletProvider>
        <WidgetWallet />
      </WalletProvider>
    );

    // Then
    expect(wallet.toJSON()).toMatchSnapshot();
    expect(wallet.root.findAllByType(Pocket).length).toEqual(3);
  });
});


// export const WidgetWalletStoryBook = () => (
//   <WalletProvider>
//     <WidgetWallet />
//   </WalletProvider>
// );
