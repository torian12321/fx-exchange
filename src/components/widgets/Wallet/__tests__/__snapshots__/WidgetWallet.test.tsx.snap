// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget: Wallet should render no items by default 1`] = `
<div
  className="css-4ey8z4-WidgetWallet"
/>
`;

exports[`Widget: Wallet should render with 3 currencies from the 'API' 1`] = `
<div
  className="css-4ey8z4-WidgetWallet"
>
  <div
    className="css-1gtg0es-Pocket"
  >
    <img
      className="css-1axo4du-Pocket"
      src="EUR.png"
    />
    <span
      className="css-3tjd33-Pocket"
    >
      1000.00 €
    </span>
  </div>
  <div
    className="css-1gtg0es-Pocket"
  >
    <img
      className="css-1axo4du-Pocket"
      src="GBP.png"
    />
    <span
      className="css-3tjd33-Pocket"
    >
      1234.00 £
    </span>
  </div>
  <div
    className="css-1gtg0es-Pocket"
  >
    <img
      className="css-1axo4du-Pocket"
      src="USD.png"
    />
    <span
      className="css-3tjd33-Pocket"
    >
      2050.00 $
    </span>
  </div>
</div>
`;
